DUI3-DX/Sdk/Speckle.Connectors.Utils/Operations/ReceiveOperation.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Speckle.Connectors.Utils.Builders;
+using Speckle.Core.Api;
+using Speckle.Core.Credentials;
+using Speckle.Core.Logging;
+using Speckle.Core.Models;
+using Speckle.Core.Transports;
+
+namespace Speckle.Connectors.Utils.Operations;
+
+public sealed class ReceiveOperation
+{
+  private readonly IHostObjectBuilder _hostObjectBuilder;
+
+  public ReceiveOperation(IHostObjectBuilder hostObjectBuilder)
+  {
+    _hostObjectBuilder = hostObjectBuilder;
+  }
+
+  public async Task<IEnumerable<string>> Execute(
+    string accountId, // POC: all these string arguments exists in ModelCard but not sure to pass this dependency here, TBD!
+    string projectId,
+    string projectName,
+    string modelName,
+    string versionId,
+    CancellationTokenSource cts,
+    Action<string, double?>? onOperationProgressed = null
+  )
+  {
+    // 2 - Check account exist
+    Account account =
+      AccountManager.GetAccounts().FirstOrDefault(acc => acc.id == accountId)
+      ?? throw new SpeckleAccountManagerException();
+
+    // 3 - Get commit object from server
+    Client apiClient = new(account);
+    ServerTransport transport = new(account, projectId);
+    Commit? version =
+      await apiClient.CommitGet(projectId, versionId, cts.Token).ConfigureAwait(false)
+      ?? throw new SpeckleException($"Failed to receive commit: {versionId} from server)");
+
+    Base? commitObject =
+      await Speckle.Core.Api.Operations
+        .Receive(version.referencedObject, cancellationToken: cts.Token, remoteTransport: transport)
+        .ConfigureAwait(false)
+      ?? throw new SpeckleException(
+        $"Failed to receive commit: {version.id} objects from server: {nameof(Operations)} returned null"
+      );
+
+    apiClient.Dispose();
```

Comment:
This type of disposal is not safe. As if an exception occurs between creating the Client, and this line, then the client will not be disposed.

Instead we should be using the `using` keyword on line 40 (where we are constructing the client)